src/statmodels.jl

```diff
@@ -233,8 +235,12 @@ function r2(obj::StatisticalModel, variant::Symbol)
         1 - exp(2 * (ll0 - ll) / nobs(obj))
     elseif variant == :Nagelkerke
         (1 - exp(2 * (ll0 - ll) / nobs(obj))) / (1 - exp(2 * ll0 / nobs(obj)))
+    elseif variant == :deviance_ratio
+        dev  = deviance(obj)
```

Comment:
This variant doesn't need `ll` and `ll0`, so you should add a top-level `if` to avoid computing these.